lib/dpul_collections_web/live/browse_live.ex

```diff
@@ -29,8 +33,31 @@ defmodule DpulCollectionsWeb.BrowseLive do
     {:noreply, push_patch(socket, to: "/browse?r=#{Enum.random(1..3000)}")}
   end
 
+  def handle_event(
+        "pin",
+        %{"item_id" => id},
+        socket = %{assigns: %{items: items, pinned_items: pinned_items}}
+      ) do
+    doc = items |> Enum.find(fn item -> item.id == id end)
+
+    if idx = Enum.find_index(pinned_items, fn pinned_item -> doc.id == pinned_item.id end) do
```

Comment:
discussion: Not blocking, but I am wondering if `case` would be more idiomatic here since we're assigning the variable idx. Something like:
```
case Enum.find_index(pinned_items, fn pinned_item -> doc.id == pinned_item.id end) do
  nil ->
      {:noreply, socket |> assign(pinned_items: Enum.concat(pinned_items, [doc]))}
  idx ->
      socket = socket |> assign(pinned_items: List.delete_at(pinned_items, idx))
      {:noreply, socket}
end

Reply:
Thank you for the suggestion! I was very unsure.